packages/editor/src/components/document-bar/index.js

```diff
@@ -42,127 +45,149 @@ const typeLabels = {
-				postType: getCurrentPostType(),
-				postId: getCurrentPostId(),
-				onNavigateToPreviousEntityRecord:
-					getSettings().onNavigateToPreviousEntityRecord,
-				getEditorSettings: getSettings,
-			};
-		},
-		[]
-	);
+const TEMPLATE_POST_TYPES = [ 'wp_template', 'wp_template_part' ];
 
-	const handleOnBack = () => {
-		if ( onNavigateToPreviousEntityRecord ) {
-			onNavigateToPreviousEntityRecord();
-		}
-	};
+const GLOBAL_POST_TYPES = [
+	...TEMPLATE_POST_TYPES,
+	'wp_block',
+	'wp_navigation',
+];
 
-	return (
-		<BaseDocumentActions
-			postType={ postType }
-			postId={ postId }
-			onBack={
-				onNavigateToPreviousEntityRecord ? handleOnBack : undefined
-			}
-		/>
-	);
-}
+const MotionButton = motion( Button );
 
-function BaseDocumentActions( { postType, postId, onBack } ) {
-	const { open: openCommandCenter } = useDispatch( commandsStore );
-	const { editedRecord: doc, isResolving } = useEntityRecord(
-		'postType',
+export default function DocumentBar() {
+	const {
 		postType,
-		postId
-	);
-	const { templateIcon, templateTitle } = useSelect( ( select ) => {
-		const { __experimentalGetTemplateInfo: getTemplateInfo } =
-			select( editorStore );
-		const templateInfo = getTemplateInfo( doc );
+		document,
+		isResolving,
+		templateIcon,
+		templateTitle,
+		onNavigateToPreviousEntityRecord,
+	} = useSelect( ( select ) => {
+		const {
+			getCurrentPostType,
+			getCurrentPostId,
+			getEditorSettings,
+			__experimentalGetTemplateInfo: getTemplateInfo,
+		} = select( editorStore );
+		const { getEditedEntityRecord, getIsResolving } = select( coreStore );
+		const _postType = getCurrentPostType();
+		const _postId = getCurrentPostId();
+		const _document = getEditedEntityRecord(
+			'postType',
+			_postType,
+			_postId
+		);
+		const _templateInfo = getTemplateInfo( _document );
 		return {
-			templateIcon: templateInfo.icon,
-			templateTitle: templateInfo.title,
+			postType: _postType,
+			document: _document,
+			isResolving: getIsResolving(
+				'getEditedEntityRecord',
+				'postType',
+				_postType,
+				_postId
+			),
+			templateIcon: _templateInfo.icon,
+			templateTitle: _templateInfo.title,
+			onNavigateToPreviousEntityRecord:
+				getEditorSettings().onNavigateToPreviousEntityRecord,
 		};
-	} );
-	const isNotFound = ! doc && ! isResolving;
-	const icon = icons[ postType ] ?? pageIcon;
-	const [ isAnimated, setIsAnimated ] = useState( false );
-	const isMounting = useRef( true );
-	const isTemplate = [ 'wp_template', 'wp_template_part' ].includes(
-		postType
-	);
-	const isGlobalEntity = [
-		'wp_template',
-		'wp_navigation',
-		'wp_template_part',
-		'wp_block',
-	].includes( postType );
+	}, [] );
 
-	useEffect( () => {
-		if ( ! isMounting.current ) {
-			setIsAnimated( true );
-		}
-		isMounting.current = false;
-	}, [ postType, postId ] );
+	const { open: openCommandCenter } = useDispatch( commandsStore );
+	const isReducedMotion = useReducedMotion();
+
+	const isNotFound = ! document && ! isResolving;
```

Comment:
Optional:

Are we only interested in whether document and/or document?.title exist?

Maybe this line could be moved into the selector, and then export `isNotFound` and `documentTitle`

Reply:
You're probably right, but I don't really want to spend more time on this 😛 These changes were supposed to be a bit of housekeeping while working on https://github.com/WordPress/gutenberg/pull/58528 but then I got a bit carried away.

Reply:
All good! 

No real performance/functionality issues.

I think the theory goes that, if we export the entire entity object, it will trigger a render whenever any of the entity's properties change, even if the title remains the same.

Nice work.